src/FixedDataTableCellGroup.js

```diff
@@ -76,25 +87,70 @@ var FixedDataTableCellGroupImpl = React.createClass({
     }, false);
 
     var currentPosition = 0;
-    for (var i = 0, j = columns.length; i < j; i++) {
+    var count = 0;
+
+    var newColumnsToRender = new Array(props.maxVisibleColumns);
+    var positions = new Array(props.maxVisibleColumns);
```

Comment:
Your right, but I think its better as an object. As an array, it'll be a sparse array.